src/components/MapComponent.vue

```diff
@@ -0,0 +1,335 @@
+    },
+
+    pushBlock: {
+      type: Function,
+      required: false,
+    },
+
+    activeBlockId: {
+      type: String,
+      required: false,
+    },
+
+    isAdminMap: {
+      type: Boolean,
+      required: false,
+    },
+  },
+
+  data: () => ({
+    labelsVisible: true,
+  }),
+
+  created() {
+    // The action button to add a block to the reserve list
+    this.addToReserve = {
+      title: 'Add',
+      id: 'reserve',
+      image: 'https://upload.wikimedia.org/wikipedia/commons/8/8f/Checkmark.svg',
+    };
+
+    // The action button to add a block to the release list
+    this.addToRelease = {
+      title: 'Release',
+      id: 'release',
+      image: 'https://upload.wikimedia.org/wikipedia/commons/e/ea/Red-subtract-icon-png-13.png',
+    };
+
+    // The action button to add a block to the complete list
+    this.addToComplete = {
+      title: 'Complete',
+      id: 'complete',
+      image: 'https://upload.wikimedia.org/wikipedia/commons/8/8f/Checkmark.svg',
+    };
+
+    // Sets the colors of blocks based on their reservation status
+    this.blockRenderer = {
+      // https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-FeatureLayer.html#renderer
+      // https://developers.arcgis.com/javascript/latest/api-reference/esri-renderers-UniqueValueRenderer.html
+      type: 'unique-value',
+      field: 'RESERVED',
+      defaultSymbol: { type: 'simple-fill' },
+      uniqueValueInfos: [
+        {
+          value: '0',
+          symbol: {
+            type: 'simple-fill', // autocasts as new SimpleFillSymbol()
+            color: 'rgba(195, 195, 195, 0.5)',
+          },
+        }, {
+          value: '1',
+          symbol: {
+            type: 'simple-fill', // autocasts as new SimpleFillSymbol()
+            color: 'rgba(240, 240, 30, 0.5)',
+          },
+        },
+        {
+          value: '2',
+          symbol: {
+            type: 'simple-fill', // autocasts as new SimpleFillSymbol()
+            color: 'rgba(134, 195, 86, 0.5)',
+          },
+        },
+      ],
+    };
+
+    // Sets the color of the marked private streets
+    this.privateStreetRenderer = {
+      type: 'simple',
+      symbol: {
+        type: 'simple-line',
+        color: 'rgba(200, 0, 0, 1)',
+        width: 2,
+      },
+    };
+
+    // The optional labels of the blocks on the map
+    this.blockLabel = {
+      labelExpressionInfo: { expression: '$feature.ID' },
+      symbol: {
+        type: 'text',
+        color: 'black',
+        haloSize: 1,
+        haloColor: 'white',
+      },
+    };
+  },
+
+  computed: {
+    ...mapState({
+      reservedBlocks: 'reservedBlocks',
+    }),
+
+    // Determines which blocks to display depending on the user is looking at
+    // reserve new blocks, current reservations, or admin map
+    blockFilter() {
+      if (this.isAdminMap) {
+        // Admins can see all current reserved blocks by any user
+        return 'RESERVED = 1 OR RESERVED = 2';
```

Comment:
🌱 I think admins would like being able to see unreserved blocks too

Reply:
Should admins be able to assign an unreserved block to themselves/any other user? Adding ether of these features could be a good ticket for the new members who are going to work with the map.

Reply:
Yeah that would be great, can you write a ticket stub? cc @florisdobber

Reply:
Yup done